pkg/trace/api/profiles.go

```diff
@@ -157,12 +164,148 @@ type multiTransport struct {
 	keys    []string
 }
 
+// isRetryableError determines if an error should trigger a retry
+func isRetryableError(err error) bool {
+	if err == nil {
+		return false
+	}
+
+	// Check for network-level errors that indicate connection issues
+	if netErr, ok := err.(net.Error); ok {
+		// Retry on timeout or temporary network errors
+		return netErr.Timeout() || netErr.Temporary()
+	}
+
+	// Check for specific connection errors
+	if opErr, ok := err.(*net.OpError); ok {
+		if opErr.Op == "dial" || opErr.Op == "read" || opErr.Op == "write" {
+			return true
+		}
+		if syscallErr, ok := opErr.Err.(*net.DNSError); ok {
+			return syscallErr.Temporary()
+		}
+		if syscallErr, ok := opErr.Err.(*syscall.Errno); ok {
+			// Common retryable syscall errors
+			switch *syscallErr {
+			case syscall.ECONNREFUSED, syscall.ECONNRESET, syscall.ETIMEDOUT, syscall.ENETUNREACH, syscall.EHOSTUNREACH:
+				return true
+			}
+		}
+	}
+
+	// Check for URL errors that might be retryable
+	if urlErr, ok := err.(*url.Error); ok {
```

Comment:
I think you can drop this check, as far as I can tell a url.Error should never be temporary as I believe these can only happen from failing to parse the provided URL